File: SDKs/googleapis.com/speech/v1p1beta1/typescript/src/sdk/models/shared/longrunningrecognizerequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RecognitionAudio } from "./recognitionaudio";
import { RecognitionConfigInput } from "./recognitionconfig";
import { TranscriptOutputConfig } from "./transcriptoutputconfig";



// LongRunningRecognizeRequestInput
/** 
 * The top-level message sent by the client for the `LongRunningRecognize` method.
**/
export class LongRunningRecognizeRequestInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=audio" })
  audio?: RecognitionAudio;

  @SpeakeasyMetadata({ data: "json, name=config" })
  config?: RecognitionConfigInput;

  @SpeakeasyMetadata({ data: "json, name=outputConfig" })
  outputConfig?: TranscriptOutputConfig;
}

File: SDKs/googleapis.com/speech/v1p1beta1/go/pkg/models/shared/longrunningrecognizerequest.go
package shared

// LongRunningRecognizeRequestInput
// The top-level message sent by the client for the `LongRunningRecognize` method.
type LongRunningRecognizeRequestInput struct {
	Audio        *RecognitionAudio       `json:"audio,omitempty"`
	Config       *RecognitionConfigInput `json:"config,omitempty"`
	OutputConfig *TranscriptOutputConfig `json:"outputConfig,omitempty"`
}
